echo-server.js

```diff
@@ -17,6 +17,8 @@ let server = net.createServer(function(connection) {
   // This tells Node what to do whenever we receive data over this connection.
   // The clientData argument contains whatever data the client sent to us.
   connection.on('data', function(clientData) {
+    console.log(`Receiving data... ${clientData}`);
```

Comment:
The `clientData` will contain the newline character `\n` from when the user hits enter. You can use `trimRight()` to remove it (`trimRight` removes trailing white space, which includes spaces, tabs, and newlines).

```javascript
let text = clientData.toString().trimRight();

console.log(`Receiving data: '${text}'`);
connection.write(text.toUpperCase());
```

Reply:
Actually, this removes the newline, so for the purposes of printing out `toUpperCase()` you might want to re-add it, like

```javascript
connection.write(text.toUpperCase() + '\n');
```